src/main/java/ch/njol/skript/classes/data/BukkitClasses.java

```diff
@@ -2020,5 +2030,209 @@ public String getVariableNamePattern() {
+						}
+					}
+				})
+				.serializer(new Serializer<BossBar>() {
+					@Override
+					public Fields serialize(BossBar bar) throws NotSerializableException {
+						if (KEYED_BOSS_BARS_EXIST && bar instanceof KeyedBossBar) {
+							Fields fields = new Fields();
+							fields.putObject("id", ((KeyedBossBar) bar).getKey().getKey());
+							return fields;
+						} else {
+							Fields fields = new Fields();
+							fields.putObject("title", bar.getTitle());
+							fields.putPrimitive("progress", bar.getProgress());
+							fields.putObject("color", bar.getColor());
+							fields.putObject("style", bar.getStyle());
+							fields.putPrimitive("visible", bar.isVisible());
+							if (bar.getPlayers().size() > 0)
+								fields.putObject("players", bar.getPlayers().toArray(new Player[0]));
+							ArrayList<BarFlag> barFlags = new ArrayList<>();
+							for (BarFlag barFlag : BarFlag.values()) {
+								if (bar.hasFlag(barFlag))
+									barFlags.add(barFlag);
+							}
+							if (barFlags.size() > 0) {
+								fields.putObject("flags", barFlags);
+							}
+							return fields;
+						}
+					}
+					
+					@Override
+					@SuppressWarnings("null")
+					protected BossBar deserialize(Fields fields) throws StreamCorruptedException, NotSerializableException {
+						if (fields.hasField("id")) {
+							BossBar bar = Bukkit.getBossBar(NamespacedKey.minecraft(fields.getAndRemoveObject("id", String.class)));
+							return bar;
+						}
+						String title = fields.getAndRemoveObject("title", String.class);
+						BarColor color = fields.getAndRemoveObject("color", BarColor.class);
+						BarStyle style = fields.getAndRemoveObject("style", BarStyle.class);
+						BossBar bar = Bukkit.createBossBar(title, color, style);
+						bar.setProgress(fields.getAndRemovePrimitive("progress", double.class));
+						bar.setVisible(fields.getAndRemovePrimitive("visible", boolean.class));
+						if (fields.hasField("flags")) {
+							for (Object flag : fields.getAndRemoveObject("flags", List.class))
+								bar.addFlag((BarFlag) flag);
+						}
+						if (fields.hasField("players")) {
+							for (OfflinePlayer player : fields.getAndRemoveObject("players", OfflinePlayer[].class)) {
+								if (player.isOnline())
+									bar.addPlayer((Player) player);
+							}
+						}
+						return bar;
+					}
+					
+					@Override
+					public void deserialize(BossBar o, Fields f) throws StreamCorruptedException, NotSerializableException {
+						assert false;
+					}
+					
+					@Override
+					public boolean mustSyncDeserialization() {
+						return true;
+					}
+					
+					@Override
+					protected boolean canBeInstantiated() {
+						return false;
+					}})
+				.changer(DefaultChangers.bossbarChanger)
+		);
+		EnumUtils<BarColor> bossbarColours = new EnumUtils<>(BarColor.class, "bossbar colors");
+		Classes.registerClass(new ClassInfo<>(BarColor.class, "bossbarcolor")
+				.user("(0|boss) ?bar colou?r")
```

Comment:
Remember, this is regex, not a Skript pattern:
```suggestion
				.user("boss ?bar colou?r")
```